packages/fungible-token/bridge-fungible-token/src/cast.sw

```diff
@@ -0,0 +1,82 @@
+library;
+
+use std::u256::U256;
+use std::constants::ZERO_B256;
+
+impl From<b256> for U256 {
+    fn from(value: b256) -> U256 {
+        let (word1, word2, word3, word4) = asm(r1: value) { r1: (u64, u64, u64, u64) };
+        let result = U256::from((word1, word2, word3, word4));
+
+        result
+    }
+
+    fn into(self) -> b256 {
+        let result: b256 = ZERO_B256;
+
+        asm(output: result, r1: self.a, r2: self.b, r3: self.c, r4: self.d) {
+            sw   output r1 i0; // store the word in r1 in output + 0 words
+            sw   output r2 i1; // store the word in r2 in output + 1 word
+            sw   output r3 i2; // store the word in r3 in output + 1 word
+            sw   output r4 i3; // store the word in r4 in output + 1 word
+        }
+
+        result
+    }
+}
+
+#[test]
+fn test_b256_addition() {
+    let one = U256::from((0, 0, 0, 1));
+    let two = U256::from((0, 0, 0, 2));
+
+    let addition = one.add(two);
+    let three = U256::from((0, 0, 0, 3));
+
+    assert(three == addition);
+
+    let three_b256: b256 = 0x0000000000000000000000000000000000000000000000000000000000000003;
+
+    assert(U256::from(three_b256) == three);
+    assert(U256::from(three_b256) == addition);
+
+    let three_into_b256: b256 = three.into();
+    assert(three_into_b256 == three_b256);
+
+    let addition_into_b256: b256 = addition.into();
+    assert(addition_into_b256 == three_b256);
+}
```

Comment:
```suggestion
fn test_b256_from() {
    // Test the boundary conditions: min, middle, max
    let min = b256::min();
    let middle = 0x000000000000000000000000000000000000000000000000000000000000000a;
    let max = b256::max();
    
    // Alternatively, compare each field of the U256
    assert_eq(U256::from(min), U256::from((0, 0, 0, 0)));
    assert_eq(U256::from(middle), U256::from((0, 0, 0, 10)));
    assert_eq(U256::from(max), U256::from((u64::max(), u64::max(), u64::max(), u64::max())));
}
```

If it complains about an import then add `use core::primitives::*;` at the top.

Test is being re-written to be a unit-test. We are not trying to test the addition of the `U256`. We are testing the conversions of the `from` & `into`